lib/Controller/Layout.php

```diff
@@ -1230,6 +1231,7 @@ function grid(Request $request, Response $response)
 
             // Populate the status message
             $layout->getStatusMessage();
+            $layout->isLocked = $this->container->get('pool')->getItem('locks/layout/' . $layout->layoutId)->get();
```

Comment:
Although we have the container here, that is actually a workaround and shouldn't be used as routine.

We should pass in the Pool as a named parameter please.

Reply:
I think you also need to check if it `isMiss()` or not, expired locks might hang around for some time after they have actually expired, depending on the cache engine and when it does garbage collection.